lckernel/cad/builders/circle.cpp

```diff
@@ -189,6 +189,13 @@ void lc::builder::CircleBuilder::modifyForTempEntity(bool val)
 
 lc::entity::Circle_CSPtr lc::builder::CircleBuilder::build()
 {
+    checkEntityConstraints();
+
+    if (_radius < 0)
```

Comment:
This check should be done in `checkEntityConstraints()`. You can override this method and call the method from the override method:
```
CADEntityBuilder::checkEntityConstraints();
```